src/other/tables.jl

```diff
@@ -43,11 +43,13 @@ fromcolumns(x, names; copycols::Bool=true) =
               copycols=copycols)
 
 function DataFrame(x::T; copycols::Bool=true) where {T}
-    if !Tables.istable(x)
-        if x isa AbstractVector && all(col -> isa(col, AbstractVector), x)
+    if !Tables.istable(x) && x isa AbstractVector && !isempty(x)
+        # here we handle the cases that are accepted by standard DataFrame constructors
+        # but eltype(x) is more flexible than assumed in these methods
```

Comment:
Exactly. What we handle is essentially `Any[...]` case (or some weird `Union{...}[...]`) as it will not be handled by standard constructors.

It was not very clear earlier and this is essentially what this code path is for only.

Reply:
OK. I think what confused me is "that are accepted by standard DataFrame constructors", since these are not really accepted, or they wouldn't have to be dealt with in this method. :-)

Reply:
Can you - as usual - propose a better wording (or what we have now is OK?). Thank you!